llrt_core/src/modules/encoding/text_encoder.rs

```diff
@@ -16,43 +17,56 @@ impl TextEncoder {
     }
 
     #[qjs(get)]
-    fn encoding(&self) -> String {
-        "utf-8".to_string()
+    fn encoding(&self) -> &str {
+        "utf-8"
     }
 
-    pub fn encode<'js>(&self, ctx: Ctx<'js>, string: String) -> Result<Value<'js>> {
-        TypedArray::new(ctx, string.as_bytes()).map(|m| m.into_value())
+    pub fn encode<'js>(&self, ctx: Ctx<'js>, string: Opt<Value<'js>>) -> Result<Value<'js>> {
+        if let Some(string) = string.0 {
+            if let Some(string) = string.as_string() {
+                if let Ok(string) = string.to_string() {
+                    return TypedArray::new(ctx, string.as_bytes()).map(|m| m.into_value());
+                }
+            }
+        }
```

Comment:
Should we really ignore errors here? If string conversion fails for instance we?

Reply:
Admittedly, I haven't read the spec yet. With the previous implementation, the `String` parameter required that the value be a string and would throw if something like `undefined` was passed in. With this change, we pass a lot more of the tests, but I can't say for sure how this is supposed to behave according to the spec. We could pause this change until I find out, but my thought was that it's just a slight improvement to what was already there. What do you think?

Reply:
Thinking about this more, maybe we should `throw` to be safe. I can see if that causes test failures and look at what behavior those tests expect.

Reply:
I was more concerned about this part:
```rust
if let Some(string) = string.as_string() {
    if let Ok(string) = string.to_string() {
        return TypedArray::new(ctx, string.as_bytes()).map(|m| m.into_value());
    }
}
```
That could be written as
```rust
if let Some(string) = string.as_string() {
    let string = string.to_string()?
    return TypedArray::new(ctx, string.as_bytes()).map(|m| m.into_value());
}
```